src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10462,20 +10466,41 @@ private ExpressionSyntax ParseTermWithoutPostfix(Precedence precedence)
                 case SyntaxKind.InterpolatedStringToken:
                     return this.ParseInterpolatedStringToken();
                 case SyntaxKind.OpenParenToken:
-                    return this.ParseCastOrParenExpressionOrLambdaOrTuple(precedence);
+                    if (IsPossibleLambdaExpression(precedence))
+                    {
+                        if (this.TryParseLambdaExpression() is { } lambda)
+                        {
+                            return lambda;
+                        }
+                    }
+                    return this.ParseCastOrParenExpressionOrTuple();
                 case SyntaxKind.NewKeyword:
                     return this.ParseNewExpression();
                 case SyntaxKind.StackAllocKeyword:
                     return this.ParseStackAllocExpression();
                 case SyntaxKind.DelegateKeyword:
+                    // check for lambda expression with explicit function pointer return type
+                    if (this.IsPossibleLambdaExpression(precedence))
+                    {
+                        return this.ParseLambdaExpression();
+                    }
                     return this.ParseAnonymousMethodExpression();
                 case SyntaxKind.RefKeyword:
+                    if (this.IsPossibleLambdaExpression(precedence))
+                    {
+                        return this.ParseLambdaExpression();
+                    }
                     // ref is not expected to appear in this position.
                     return this.AddError(ParsePossibleRefExpression(), ErrorCode.ERR_InvalidExprTerm, SyntaxFacts.GetText(tk));
                 default:
                     // check for intrinsic type followed by '.'
                     if (IsPredefinedType(tk))
                     {
+                        if (this.IsPossibleLambdaExpression(precedence))
+                        {
+                            return this.ParseLambdaExpression();
+                        }
+
```

Comment:
up to you, but after having read this, i'm strongly on the fence about extracting this code up higher prior to the main switch, and just having it be a short switch dedicated to the "is this a token that could start a lambda?  ok, see if it's a lambda with this precedence.  Ok now parse out lambda".  Having itt be so scattershot over all the cases makes it hard to read the code int he entirety and know what's going on. 

but again, your call here if that change is desirable.

Reply:
> i'm strongly on the fence about extracting this code up higher prior to the main switch, and just having it be a short switch dedicated to the "is this a token that could start a lambda?

Let me keep that in mind as a potential refactoring for later. It would reduce the number of cases where we check for lambdas, but it might increase the complexity of `IsPossibleLambdaExpression()`.

Reply:
works for me.  i could go either way here.